src/Features/CSharp/Portable/CodeFixes/ConditionalExpressionInStringInterpolation/CSharpAddParenthesesAroundConditionalExpressionInInterpolatedStringCodeFixProvider.cs

```diff
@@ -0,0 +1,80 @@
+                    cancellationToken => GetChangedDocumentAsync(context.Document, conditionalExpression.SpanStart, cancellationToken));
+                context.RegisterCodeFix(documentChangeAction, diagnostic);
+            }
+        }
+
+        private static async Task<Document> GetChangedDocumentAsync(Document document, int conditionalExpressionSyntaxStartPosition, CancellationToken cancellationToken)
+        {
+            // The usual SyntaxTree transformations are complicated if string literals are present in the false part as in
+            // $"{ condition ? "Success": "Failure" }"
+            // The colon starts a FormatClause and the double quote left to 'F' therefore ends the interpolated string.
+            // The text starting with 'F' is parsed as code and the resulting syntax tree is impractical.
+            // The same problem arises if a } is present in the false part.
+            // To circumvent these problems this solution
+            // 1. Inserts an opening parenthesis
+            // 2. Re-parses the resulting document (now the colon isn't treated as starting a FormatClause anymore)
+            // 3. Replaces the missing CloseParenToken with a new one
+            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var openParenthesisPosition = conditionalExpressionSyntaxStartPosition;
+            var textWithOpenParenthesis = text.Replace(openParenthesisPosition, 0, "(");
+            var documentWithOpenParenthesis = document.WithText(textWithOpenParenthesis);
+            var syntaxRoot = await documentWithOpenParenthesis.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var nodeAtInsertPosition = syntaxRoot.FindNode(new TextSpan(openParenthesisPosition, 0));
+            if (nodeAtInsertPosition is ParenthesizedExpressionSyntax parenthesizedExpression &&
+                parenthesizedExpression.CloseParenToken.IsMissing)
+            {
+                var parenthesizedExpressionWithClosingParen =
+                    parenthesizedExpression.WithCloseParenToken(SyntaxFactory.Token(SyntaxKind.CloseParenToken)).
+                    WithTriviaFrom(parenthesizedExpression);
+                syntaxRoot = syntaxRoot.ReplaceNode(parenthesizedExpression, parenthesizedExpressionWithClosingParen);
+                return documentWithOpenParenthesis.WithSyntaxRoot(syntaxRoot);
+            }
+
+            return documentWithOpenParenthesis;
+        }
+
+        private class MyCodeAction : CodeAction.DocumentChangeAction
+        {
+            public MyCodeAction(Func<CancellationToken, Task<Document>> createChangedDocument)
+                : base(
+                      CSharpFeaturesResources.Add_parentheses_around_conditional_expression_in_interpolated_string,
+                      createChangedDocument,
+                      CSharpFeaturesResources.Add_parentheses_around_conditional_expression_in_interpolated_string)
```

Comment:
nit, our convention is to keep the first argument with base(CSharp..., and then align the other args with that first one if they wrap.

But don't need to fix if you don't want to.